ConsoleApp/Stack.cs

```diff
@@ -0,0 +1,65 @@
+using System;
+using static System.Console;
+namespace ConsoleApp;
+
+class Stack<T>
+{
+    private record ListItem
+    {
+        public T Value { get; init; }
+        public ListItem Previous { get; set; }
+    }
+
+    private ListItem _last;
+    public int Height { get; private set; }
+
+    public void Push(T value)
+    {
+
+        var newItem = new ListItem
+        {
+            Value = value,
+        };
+
+        if (_last != null) newItem.Previous = _last;
+
+        _last = newItem;
+        Height++;
+
+    }
+
+    public T Pop()
+    {
+        if (_last == null) throw new NullReferenceException();
+
+        T value = _last.Value;
+        _last = _last.Previous;
+        Height--;
+        return value;
+    }
+
+    public void Clear()
+    {
+        if (_last == null) throw new NullReferenceException();
```

Comment:
it's not needed, if stack is empty - do nothing

Reply:
fixed